File: src/main/java/com/group/libraryapp/domain/book/Book.java
package com.group.libraryapp.domain.book;

import javax.persistence.Column;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;

public class Book {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id = null;

    @Column(nullable = false)
    private String name;

    protected Book() {
    }

    public Book(String name) {
        if (name == null || name.isBlank()) {
            throw new IllegalArgumentException(String.format("error: name(%s) is not exist", name));
        }
        this.name = name;
    }

    public String getName() {
        return name;
    }
}

File: src/main/kotlin/com/group/libraryapp/domain/book/Book.kt
package com.group.libraryapp.domain.book

import javax.persistence.*

@Entity
class Book(
  val name: String,

  @Enumerated(EnumType.STRING)
  val type: BookType,

  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  val id: Long? = null,
) {

  init {
    if (name.isBlank()) {
      throw IllegalArgumentException("Name not allowed blank.")
    }
  }

  companion object {
    fun fixture(
      name: String = "Book Name",
      type: BookType = BookType.COMPUTER,
      id: Long? = null,
    ): Book {
      return Book(
        name = name,
        type = type,
        id = id,
      )
    }
  }
}